### PR TITLE
feat: add STRDIAG010 docs

### PR DESCRIPTION
**Pull request by IXLLEGACYIXL**
Adds documentation for the new analyzer https://github.com/stride3d/stride/pull/2104
should probably be released with the release of the next stride version which contains the analyzer  as the analyzer links to this site

### CODE REVIEW

**Comment by IXLLEGACYIXL** [comment]
the PR is merged so this can be merged as well after review

File: en/diagnostics/STRDIAG010.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Diagnostics Warning STRDIAG010
+
+> The Type '{0}' doesn't have a public parameterless constructor, which is needed for Serialization
```
Comment (VaclavElias) [inline en/diagnostics/STRDIAG010.md:3]: Full stop at the end of the sentece. Just to keep the consistency with the other STRDIAG pages intros.

File: en/diagnostics/STRDIAG010.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+## Explanation
+
+By default, Stride serializers only support types that have a parameterless constructor.
```
Comment (VaclavElias) [inline en/diagnostics/STRDIAG010.md:7]: Make sure that each sentence (4 sentences in this case) follows the next one without a new line break. Ignore if I am wrong 🤣.

**Review by VaclavElias — CHANGES_REQUESTED** [review]
Just some consistency updates :)

File: en/diagnostics/STRDIAG010.md (hunk 2 above)
Comment (IXLLEGACYIXL) [reply en/diagnostics/STRDIAG010.md]: did that

**Review by VaclavElias — APPROVED** [review]

**Comment by VaclavElias** [comment]
Thank you @IXLLEGACYIXL.

**Pull request merged**